### PR TITLE
fix(nx): npm scripts documentation generation

### LINKED ISSUE #1567: npm scripts documentation is incorrect
## Expected Behavior

The npm scripts documentation should list only available options for each script/command.

## Current Behavior

The npm scripts documentation lists options that aren't available for scripts/commands. For example, many of the options listed in the [workspace schematic doc](https://github.com/nrwl/nx/blob/master/docs/api-workspace/npmscripts/workspace-schematic-name.md) aren't and shouldn't be available to that script/command.

### PR DESCRIPTION
**Pull request by BuckyMaler**
Before, a single instance of `yargs` was accumulating options, so commands parsed later would have options from prior commands. That's because all `yargs` options are [global by default](http://yargs.js.org/docs/#api-globalglobals-globaltrue). Now, a new instance of `yargs` is required before each command is parsed, so global options aren't accumulated.

I initially added `global: false` to all options in `nx-commands.ts`, but that didn't work because some `yargs` methods such as [implies](https://github.com/nrwl/nx/blob/master/packages/workspace/src/command-line/nx-commands.ts#L189) set globals. Since options being global doesn't seem to be an issue except with the docs gen, I chose to make the change there.

👉 `lint-files`, `update`, `update:check`, and `update:skip` now only document `help` and `version` because the docs are based on `yargs` options and those commands don't have specific `yargs` options. How would you like to handle these? We can leave them as is, filter them out, or add code to the docs gen to handle them? Removing them makes sense to me since `lint-files` and `update:skip` aren't npm scripts, and `update` and `update:check` are handled by Angular CLI.

fixes #1567 #1535

### CODE REVIEW

File: docs/api-workspace/npmscripts/affected-e2e.md [hunk 1]
```diff
@@ -36,26 +36,12 @@ Latest commit of the current branch (usually HEAD)
 
 Show help
 
-### maxParallel
```
Comment (vsavkin) [inline docs/api-workspace/npmscripts/affected-e2e.md]: why is this removed? all affected commands should have this option?

**Review by vsavkin — CHANGES_REQUESTED** [review]

**Comment by vsavkin** [comment]
> update, update:check, and update:skip
I'd leave them as is

File: docs/api-workspace/npmscripts/affected-e2e.md (hunk 1 above)
Comment (BuckyMaler) [reply docs/api-workspace/npmscripts/affected-e2e.md]: The parallel options aren't added for the [affected:e2e](https://github.com/nrwl/nx/blob/master/packages/workspace/src/command-line/nx-commands.ts#L71) command. I assumed there was reason, but I'm not sure why myself.

Comment (BuckyMaler) [reply docs/api-workspace/npmscripts/affected-e2e.md]: I think this is a mistake. Would you like me to add the parallel options?

Comment (vsavkin) [reply docs/api-workspace/npmscripts/affected-e2e.md]: Yup. All the affected command support the parallel option, so it should be reflected in the docs.

Comment (BuckyMaler) [reply docs/api-workspace/npmscripts/affected-e2e.md]: There's a port in use error when running `affected:e2e` in parallel.

Screen Shot 2019-07-17 at 11 03 03 PM

### Steps to Reproduce

1. Generate a workspace with a single Angular app `npm init nx-workspace myworkspace`
2. Generate a second Angular app `ng g app my-app`
3. `npm run affected:e2e -- --all --headless --parallel`

Ideally the builder would handle this by setting port to 0, but I'm having trouble determining when to do that. The obvious condition is when `parallel` is true, but that option is always false in the builder regardless of whether you specify it as true on the command line because it's [filtered out](https://github.com/nrwl/nx/blob/master/packages/workspace/src/command-line/affected.ts#L55) before being passed to the builder.

Would it be alright if this is handled as a separate issue and this PR can be merged as is?

**Comment by vsavkin** [comment]
Rebased and merged it here: https://github.com/nrwl/nx/pull/1690

Thank you!
